src/http/request.cr

```diff
@@ -98,6 +98,19 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    host = @headers["Host"]?
+    return unless host
+    index = host.index(":")
+    index ? host[0...index] : host
```

Comment:
I think we can consider the index of `]` first, and compute the index of `:` after it if it's there, and then split.